engine/src/main/java/org/terasology/rendering/dag/ConditionDependentNode.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.api.client.util.Lists;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * TODO: Add javadocs
+ */
+public abstract class ConditionDependentNode extends AbstractNode implements PropertyChangeListener {
```

Comment:
This is a tricky class I find. I just understood that it is written this way to allow for multiple conditions. I'm not sure if we have a use case for that already but... ok. I guess the one problem I have is with the term "Condition". It's quite generic while the class really only deals with boolean values. Not only that: all conditions have to be true. So, a node that, say, needs to be inserted into the graph if the condition is false (for whatever reason) can't use this.

So, it seems to me that the usability of this class is quite narrow. I'd remove it in favor of nodes subscribing to the rendering config on initialization and implementing the PropertyChangeListener interface, to leave maximum freedom to the node developers.